polaris/ocean/tasks/barotropic_gyre/forward.py

```diff
@@ -0,0 +1,178 @@
+            tasks as long as this is not below ``min_tasks``
+
+        min_tasks : int, optional
+            the number of tasks the step requires.  If the system has fewer
+            than this number of tasks, the step will fail
+
+        openmp_threads : int, optional
+            the number of OpenMP threads the step will use
+
+        run_time_steps : int or None
+            Number of time steps to run for
+
+        graph_target : str, optional
+            The graph file name (relative to the base work directory).
+            If none, it will be created.
+        """
+        self.run_time_steps = run_time_steps
+        super().__init__(component=component, name=name, subdir=subdir,
+                         indir=indir, ntasks=ntasks, min_tasks=min_tasks,
+                         openmp_threads=openmp_threads,
+                         graph_target=graph_target)
+
+        # make sure output is double precision
+        self.add_yaml_file('polaris.ocean.config', 'output.yaml')
+
+        self.add_input_file(filename='init.nc',
+                            target='../init/init.nc')
+        self.add_input_file(filename='forcing.nc',
+                            target='../init/forcing.nc')
+
+        self.add_output_file(
+            filename='output.nc',
+            validate_vars=['layerThickness', 'normalVelocity'])
+
+        self.package = 'polaris.ocean.tasks.barotropic_gyre'
+        self.yaml_filename = 'forward.yaml'
+
+    def compute_cell_count(self):
+        """
+        Compute the approximate number of cells in the mesh, used to constrain
+        resources
+
+        Returns
+        -------
+        cell_count : int or None
+            The approximate number of cells in the mesh
+        """
+        section = self.config['barotropic_gyre']
+        lx = section.getfloat('lx')
+        resolution = section.getfloat('resolution')
+        ly = section.getfloat('ly')
+        nx, ny = compute_planar_hex_nx_ny(lx, ly, resolution)
+        cell_count = nx * ny
+        return cell_count
+
+    def dynamic_model_config(self, at_setup):
+        """
+        Add model config options, namelist, streams and yaml files using config
+        options or template replacements that need to be set both during step
+        setup and at runtime
+
+        Parameters
+        ----------
+        at_setup : bool
+            Whether this method is being run during setup of the step, as
+            opposed to at runtime
+        """
+        super().dynamic_model_config(at_setup)
+
+        config = self.config
+
+        model = config.get('ocean', 'model')
+        if model == 'mpas-ocean':
+            self.add_yaml_file('polaris.ocean.config', 'single_layer.yaml')
+
+        nu = config.getfloat("barotropic_gyre", "nu_2")
+        rho_0 = config.getfloat("barotropic_gyre", "rho_0")
+
+        dt_max = compute_max_time_step(config)
+        dt = dt_max / 3.
```

Comment:
It seems like we're getting `dt_max = 250` and `dt = 250/3.`:
```
config_dt = '0000_00:01:23.333'
```
I realize things are working but I wonder if it's worth having a time steps that's an even number of seconds.